workerpool.go

```diff
@@ -170,27 +170,15 @@ var workerChanCap = func() int {
 
 func (wp *workerPool) getCh() *workerChan {
 	var ch *workerChan
-	createWorker := false
-
-	wp.lock.Lock()
-	ready := wp.ready
-	n := len(ready) - 1
-	if n < 0 {
-		if wp.workersCount < wp.MaxWorkersCount {
-			createWorker = true
-			wp.workersCount++
-		}
-	} else {
-		ch = ready[n]
-		ready[n] = nil
-		wp.ready = ready[:n]
+	var createWorker bool
+
+	ch = wp.ready.pop()
+	if ch == nil && atomic.LoadInt32(&wp.workersCount) < int32(wp.MaxWorkersCount) {
```

Comment:
We might need atomic.CompareAndSwapInt32 here to avoid the momentary change of wp.workersCount.

Reply:
```go

if ch == nil {
		for {
			currentworkers := atomic.LoadInt32(&wp.workersCount)
			if currentworkers >= int32(wp.MaxWorkersCount) {
				break
			}
			if atomic.CompareAndSwapInt32(&wp.workersCount, currentworkers, currentworkers+1) {
				createWorker.Store(true)
				break
			}
		}
	} 
	
	
	
	I think this approach might help solve the problem?